sys/net/gnrc/netapi/gnrc_netapi.c

```diff
@@ -94,54 +94,52 @@ static inline int _snd_rcv_mbox(mbox_t *mbox, uint16_t type, gnrc_pktsnip_t *pkt
 int gnrc_netapi_dispatch(gnrc_nettype_t type, uint32_t demux_ctx,
                          uint16_t cmd, gnrc_pktsnip_t *pkt)
 {
-    int numof = gnrc_netreg_num(type, demux_ctx);
+    int numof = 0;
+    gnrc_netreg_entry_t *sendto = gnrc_netreg_lookup(type, demux_ctx);
 
-    if (numof != 0) {
-        gnrc_netreg_entry_t *sendto = gnrc_netreg_lookup(type, demux_ctx);
-
-        gnrc_pktbuf_hold(pkt, numof - 1);
-
-        while (sendto) {
+    while (sendto) {
+        if (numof != 0) {
+            gnrc_pktbuf_hold(pkt, 1);
```

Comment:
I just realized that this is breaking GNRC: `gnrc_pktbuf_hold()` is supposed to be called with the number of threads waiting for `pkt` of type `type` *before* is it sent to any other thread (otherwise write protection can't work properly, since `gnrc_pktbuf_start_write()` might run into a race condition, were hold wasn't called here yet). So we need to count the threads beforehand => `gnrc_netreg_num()` is required here => unless there is a better plan, I revert for now.